src/Playwright/Helpers/UrlHelpers.cs

```diff
@@ -0,0 +1,62 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Playwright.Helpers
+{
+    internal static class UrlHelpers
+    {
+        internal static bool UrlMatches(string baseURL, string urlString, string matchString, Regex matchRegex = null, Func<string, bool> matchFunction = null)
+        {
+            if (string.IsNullOrEmpty(matchString) && matchRegex == null && matchFunction == null)
+                return true;
+            if (!string.IsNullOrEmpty(matchString) && !matchString.StartsWith("*", System.StringComparison.Ordinal))
+                matchString = ConstructURLBasedOnBaseURL(baseURL, matchString);
+            if (!string.IsNullOrEmpty(matchString))
+                matchRegex = new Regex(matchString.GlobToRegex());
+            if (matchRegex != null)
+                return matchRegex.IsMatch(urlString);
+            if (!string.IsNullOrEmpty(matchString) && matchString == urlString)
+                return true;
+            var url = new Uri(urlString);
+            if (url == null)
+                return false;
```

Comment:
I don't think that this check would ever be `true`.
```suggestion
```

Also it would be nice the `Uri` is only created inside of `if (!string.IsNullOrEmpty(matchString))` (the next line) where it gets used.

Reply:
ack